Submitted By: Robert Connolly <robert at linuxfromscratch dot org> (ashes)
Date: 2007-02-14
Initial Package Version: 4.1.2
Upstream Status: Not Sent - LFS Specific
Origin: Idea originally developed by Ryan Oliver and Greg Schafer for
        the Pure LFS project.
        More architectures added by Zack Winkles.
        Further fine tunings by Greg Schafer.
        Rediffed against gcc 4.0.0 by Robert Connolly.
        Rediffed against gcc 4.1.0 by Chris Staub
        Rediffed against gcc 4.1.2 by Jim Gifford
        Rediffed against gcc 4.2.0 by Joe Ciccone
Description: This patch modifies the location of the dynamic linker for
        the GCC Pass 2 build in LFS Chapter 5.

diff -Naur gcc-4.2.0.orig/gcc/config/alpha/gnu.h gcc-4.2.0/gcc/config/alpha/gnu.h
--- gcc-4.2.0.orig/gcc/config/alpha/gnu.h	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/alpha/gnu.h	2007-05-18 16:27:10.000000000 -0400
@@ -11,7 +11,7 @@
     } while (0)
 
 #undef ELF_DYNAMIC_LINKER
-#define ELF_DYNAMIC_LINKER	"/lib/ld.so"
+#define ELF_DYNAMIC_LINKER	"/tools/lib/ld.so"
 
 #undef	STARTFILE_SPEC
 #define STARTFILE_SPEC \
diff -Naur gcc-4.2.0.orig/gcc/config/alpha/linux-elf.h gcc-4.2.0/gcc/config/alpha/linux-elf.h
--- gcc-4.2.0.orig/gcc/config/alpha/linux-elf.h	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/alpha/linux-elf.h	2007-05-18 16:23:31.000000000 -0400
@@ -28,8 +28,8 @@
 #define SUBTARGET_EXTRA_SPECS \
 { "elf_dynamic_linker", ELF_DYNAMIC_LINKER },
 
-#define GLIBC_DYNAMIC_LINKER	"/lib/ld-linux.so.2"
-#define UCLIBC_DYNAMIC_LINKER "/lib/ld-uClibc.so.0"
+#define GLIBC_DYNAMIC_LINKER	"/tools/lib/ld-linux.so.2"
+#define UCLIBC_DYNAMIC_LINKER "/tools/lib/ld-uClibc.so.0"
 #if UCLIBC_DEFAULT
 #define CHOOSE_DYNAMIC_LINKER(G, U) "%{mglibc:%{muclibc:%e-mglibc and -muclibc used together}" G ";:" U "}"
 #else
diff -Naur gcc-4.2.0.orig/gcc/config/arm/linux-eabi.h gcc-4.2.0/gcc/config/arm/linux-eabi.h
--- gcc-4.2.0.orig/gcc/config/arm/linux-eabi.h	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/arm/linux-eabi.h	2007-05-18 16:26:45.000000000 -0400
@@ -53,7 +53,7 @@
 /* Use ld-linux.so.3 so that it will be possible to run "classic"
    GNU/Linux binaries on an EABI system.  */
 #undef GLIBC_DYNAMIC_LINKER
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.3"
+#define GLIBC_DYNAMIC_LINKER "/tools/lib/ld-linux.so.3"
 
 /* At this point, bpabi.h will have clobbered LINK_SPEC.  We want to
    use the GNU/Linux version, not the generic BPABI version.  */
diff -Naur gcc-4.2.0.orig/gcc/config/arm/linux-elf.h gcc-4.2.0/gcc/config/arm/linux-elf.h
--- gcc-4.2.0.orig/gcc/config/arm/linux-elf.h	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/arm/linux-elf.h	2007-05-18 16:23:31.000000000 -0400
@@ -51,7 +51,7 @@
 
 #define LIBGCC_SPEC "%{msoft-float:-lfloat} %{mfloat-abi=soft*:-lfloat} -lgcc"
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER "/tools/lib/ld-linux.so.2"
 
 #define LINUX_TARGET_LINK_SPEC  "%{h*} %{version:-v} \
    %{b} \
diff -Naur gcc-4.2.0.orig/gcc/config/cris/linux.h gcc-4.2.0/gcc/config/cris/linux.h
--- gcc-4.2.0.orig/gcc/config/cris/linux.h	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/cris/linux.h	2007-05-18 16:30:54.000000000 -0400
@@ -76,7 +76,7 @@
 #undef CRIS_SUBTARGET_VERSION
 #define CRIS_SUBTARGET_VERSION " - cris-axis-linux-gnu"
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER "/tools/lib/ld.so.1"
 
 /* We need an -rpath-link to ld.so.1, and presumably to each directory
    specified with -B.  */
dff -Naur gcc-4.2.0.orig/gcc/config/frv/linux.h gcc-4.2.0/gcc/config/frv/linux.h
--- gcc-4.2.0.orig/gcc/config/frv/linux.h	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/frv/linux.h	2007-05-18 16:23:31.000000000 -0400
@@ -36,7 +36,7 @@
 #define ENDFILE_SPEC \
   "%{shared|pie:crtendS.o%s;:crtend.o%s} crtn.o%s"
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER "/tools/lib/ld.so.1"
 
 #undef LINK_SPEC
 #define LINK_SPEC "\
diff -Naur gcc-4.2.0.orig/gcc/config/i386/gnu.h gcc-4.2.0/gcc/config/i386/gnu.h
--- gcc-4.2.0.orig/gcc/config/i386/gnu.h	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/i386/gnu.h	2007-05-18 16:23:31.000000000 -0400
@@ -22,7 +22,7 @@
   %{!shared: \
     %{!static: \
       %{rdynamic:-export-dynamic} \
-      %{!dynamic-linker:-dynamic-linker /lib/ld.so}} \
+      %{!dynamic-linker:-dynamic-linker /tools/lib/ld.so}} \
     %{static:-static}}"
 
 #undef	STARTFILE_SPEC
diff -Naur gcc-4.2.0.orig/gcc/config/i386/linux64.h gcc-4.2.0/gcc/config/i386/linux64.h
--- gcc-4.2.0.orig/gcc/config/i386/linux64.h	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/i386/linux64.h	2007-05-18 16:23:31.000000000 -0400
@@ -49,8 +49,8 @@
    When the -shared link option is used a final link is not being
    done.  */
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux-x86-64.so.2"
+#define GLIBC_DYNAMIC_LINKER32 "/tools/lib32/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER64 "/tools/lib/ld-linux-x86-64.so.2"
 
 #undef	LINK_SPEC
 #define LINK_SPEC "%{!m32:-m elf_x86_64} %{m32:-m elf_i386} \
diff -Naur gcc-4.2.0.orig/gcc/config/i386/linux.h gcc-4.2.0/gcc/config/i386/linux.h
--- gcc-4.2.0.orig/gcc/config/i386/linux.h	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/i386/linux.h	2007-05-18 16:23:31.000000000 -0400
@@ -101,7 +101,7 @@
 
 /* These macros may be overridden in k*bsd-gnu.h and i386/k*bsd-gnu.h. */
 #define LINK_EMULATION "elf_i386"
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER "/tools/lib/ld-linux.so.2"
 
 #undef  SUBTARGET_EXTRA_SPECS
 #define SUBTARGET_EXTRA_SPECS \
diff -Naur gcc-4.2.0.orig/gcc/config/ia64/linux.h gcc-4.2.0/gcc/config/ia64/linux.h
--- gcc-4.2.0.orig/gcc/config/ia64/linux.h	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/ia64/linux.h	2007-05-18 16:23:31.000000000 -0400
@@ -37,7 +37,7 @@
 /* Define this for shared library support because it isn't in the main
    linux.h file.  */
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux-ia64.so.2"
+#define GLIBC_DYNAMIC_LINKER "/tools/lib/ld-linux-ia64.so.2"
 
 #undef LINK_SPEC
 #define LINK_SPEC "\
diff -Naur gcc-4.2.0.orig/gcc/config/linux.h gcc-4.2.0/gcc/config/linux.h
--- gcc-4.2.0.orig/gcc/config/linux.h	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/linux.h	2007-05-18 16:23:31.000000000 -0400
@@ -112,9 +112,9 @@
    GLIBC_DYNAMIC_LINKER must be defined for each target using them, or
    GLIBC_DYNAMIC_LINKER32 and GLIBC_DYNAMIC_LINKER64 for targets
    supporting both 32-bit and 64-bit compilation.  */
-#define UCLIBC_DYNAMIC_LINKER "/lib/ld-uClibc.so.0"
-#define UCLIBC_DYNAMIC_LINKER32 "/lib/ld-uClibc.so.0"
-#define UCLIBC_DYNAMIC_LINKER64 "/lib/ld64-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKER "/tools/lib/ld-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKER32 "/tools/lib/ld-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKER64 "/tools/lib/ld64-uClibc.so.0"
 #define LINUX_DYNAMIC_LINKER \
   CHOOSE_DYNAMIC_LINKER (GLIBC_DYNAMIC_LINKER, UCLIBC_DYNAMIC_LINKER)
 #define LINUX_DYNAMIC_LINKER32 \
@@ -127,3 +127,7 @@
 #define TARGET_C99_FUNCTIONS (OPTION_GLIBC)
 
 #define TARGET_POSIX_IO
+
+/* Remove /usr/include from the end of the include search path.  */
+#undef STANDARD_INCLUDE_DIR
+#define STANDARD_INCLUDE_DIR 0
diff -Naur gcc-4.2.0.orig/gcc/config/m32r/linux.h gcc-4.2.0/gcc/config/m32r/linux.h
--- gcc-4.2.0.orig/gcc/config/m32r/linux.h	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/m32r/linux.h	2007-05-18 16:23:31.000000000 -0400
@@ -59,7 +59,7 @@
 
 /* If ELF is the default format, we should not use /lib/elf.  */
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER "/tools/lib/ld-linux.so.2"
 
 #undef	LINK_SPEC
 #if TARGET_LITTLE_ENDIAN
diff -Naur gcc-4.2.0.orig/gcc/config/m68k/linux.h gcc-4.2.0/gcc/config/m68k/linux.h
--- gcc-4.2.0.orig/gcc/config/m68k/linux.h	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/m68k/linux.h	2007-05-18 16:23:31.000000000 -0400
@@ -123,7 +123,7 @@
 
 /* If ELF is the default format, we should not use /lib/elf.  */
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER "/tools/lib/ld.so.1"
 
 #undef	LINK_SPEC
 #define LINK_SPEC "-m m68kelf %{shared} \
diff -Naur gcc-4.2.0.orig/gcc/config/mips/linux64.h gcc-4.2.0/gcc/config/mips/linux64.h
--- gcc-4.2.0.orig/gcc/config/mips/linux64.h	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/mips/linux64.h	2007-05-18 16:23:31.000000000 -0400
@@ -36,10 +36,10 @@
 %{!shared: %{pthread:-lpthread} \
   %{profile:-lc_p} %{!profile: -lc}}"
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld.so.1"
-#define GLIBC_DYNAMIC_LINKERN32 "/lib32/ld.so.1"
-#define UCLIBC_DYNAMIC_LINKERN32 "/lib32/ld-uClibc.so.0"
+#define GLIBC_DYNAMIC_LINKER32 "/tools/lib32/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER64 "/tools/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKERN32 "/tools/lib64/ld.so.1"
+#define UCLIBC_DYNAMIC_LINKERN32 "/tools/lib64/ld-uClibc.so.0"
 #define LINUX_DYNAMIC_LINKERN32 \
   CHOOSE_DYNAMIC_LINKER (GLIBC_DYNAMIC_LINKERN32, UCLIBC_DYNAMIC_LINKERN32)
 
diff -Naur gcc-4.2.0.orig/gcc/config/mips/linux.h gcc-4.2.0/gcc/config/mips/linux.h
--- gcc-4.2.0.orig/gcc/config/mips/linux.h	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/mips/linux.h	2007-05-18 16:23:31.000000000 -0400
@@ -104,7 +104,7 @@
 #undef MIPS_DEFAULT_GVALUE
 #define MIPS_DEFAULT_GVALUE 0
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER "/tools/lib/ld.so.1"
 
 /* Borrowed from sparc/linux.h */
 #undef LINK_SPEC
diff -Naur gcc-4.2.0.orig/gcc/config/mn10300/linux.h gcc-4.2.0/gcc/config/mn10300/linux.h
--- gcc-4.2.0.orig/gcc/config/mn10300/linux.h	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/mn10300/linux.h	2007-05-18 16:23:31.000000000 -0400
@@ -33,7 +33,7 @@
 #undef  ASM_SPEC
 #define ASM_SPEC "%{Wa,*:%*}"
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER "/tools/lib/ld.so.1"
 
 #undef  LINK_SPEC
 #define LINK_SPEC "%{mrelax:--relax} %{shared:-shared} \
diff -Naur gcc-4.2.0.orig/gcc/config/pa/pa-linux.h gcc-4.2.0/gcc/config/pa/pa-linux.h
--- gcc-4.2.0.orig/gcc/config/pa/pa-linux.h	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/pa/pa-linux.h	2007-05-18 16:23:31.000000000 -0400
@@ -45,7 +45,7 @@
 /* Define this for shared library support because it isn't in the main
    linux.h file.  */
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER "/tools/lib/ld.so.1"
 
 #undef LINK_SPEC
 #define LINK_SPEC "\
diff -Naur gcc-4.2.0.orig/gcc/config/rs6000/linux64.h gcc-4.2.0/gcc/config/rs6000/linux64.h
--- gcc-4.2.0.orig/gcc/config/rs6000/linux64.h	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/rs6000/linux64.h	2007-05-18 16:23:31.000000000 -0400
@@ -336,10 +336,10 @@
 #undef	LINK_OS_DEFAULT_SPEC
 #define LINK_OS_DEFAULT_SPEC "%(link_os_linux)"
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld64.so.1"
-#define UCLIBC_DYNAMIC_LINKER32 "/lib/ld-uClibc.so.0"
-#define UCLIBC_DYNAMIC_LINKER64 "/lib/ld64-uClibc.so.0"
+#define GLIBC_DYNAMIC_LINKER32 "/tools/lib32/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER64 "/tools/lib/ld64.so.1"
+#define UCLIBC_DYNAMIC_LINKER32 "/tools/lib/ld-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKER64 "/tools/lib/ld64-uClibc.so.0"
 #if UCLIBC_DEFAULT
 #define CHOOSE_DYNAMIC_LINKER(G, U) "%{mglibc:%{muclibc:%e-mglibc and -muclibc used together}" G ";:" U "}"
 #else
diff -Naur gcc-4.2.0.orig/gcc/config/rs6000/sysv4.h gcc-4.2.0/gcc/config/rs6000/sysv4.h
--- gcc-4.2.0.orig/gcc/config/rs6000/sysv4.h	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/rs6000/sysv4.h	2007-05-18 16:29:05.000000000 -0400
@@ -892,8 +892,8 @@
 
 #define LINK_START_LINUX_SPEC ""
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
-#define UCLIBC_DYNAMIC_LINKER "/lib/ld-uClibc.so.0"
+#define GLIBC_DYNAMIC_LINKER "/tools/lib/ld.so.1"
+#define UCLIBC_DYNAMIC_LINKER "/tools/lib/ld-uClibc.so.0"
 #if UCLIBC_DEFAULT
 #define CHOOSE_DYNAMIC_LINKER(G, U) "%{mglibc:%{muclibc:%e-mglibc and -muclibc used together}" G ";:" U "}"
 #else
@@ -935,7 +935,7 @@
 
 #define LINK_OS_GNU_SPEC "-m elf32ppclinux %{!shared: %{!static: \
   %{rdynamic:-export-dynamic} \
-  %{!dynamic-linker:-dynamic-linker /lib/ld.so.1}}}"
+  %{!dynamic-linker:-dynamic-linker /tools/lib/ld.so.1}}}"
 
 #define CPP_OS_GNU_SPEC "-D__unix__ -D__gnu_hurd__ -D__GNU__	\
 %{!undef:					                \
diff -Naur gcc-4.2.0.orig/gcc/config/s390/linux.h gcc-4.2.0/gcc/config/s390/linux.h
--- gcc-4.2.0.orig/gcc/config/s390/linux.h	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/s390/linux.h	2007-05-18 16:23:31.000000000 -0400
@@ -73,8 +73,8 @@
 #define MULTILIB_DEFAULTS { "m31" }
 #endif
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
-#define GLIBC_DYNAMIC_LINKER64 "/lib/ld64.so.1"
+#define GLIBC_DYNAMIC_LINKER32 "/tools/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER64 "/tools/lib/ld64.so.1"
 
 #undef  LINK_SPEC
 #define LINK_SPEC \
diff -Naur gcc-4.2.0.orig/gcc/config/sh/linux.h gcc-4.2.0/gcc/config/sh/linux.h
--- gcc-4.2.0.orig/gcc/config/sh/linux.h	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/sh/linux.h	2007-05-18 16:23:31.000000000 -0400
@@ -48,7 +48,7 @@
 
 #define TARGET_ASM_FILE_END file_end_indicate_exec_stack
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER "/tools/lib/ld-linux.so.2"
 
 #undef SUBTARGET_LINK_EMUL_SUFFIX
 #define SUBTARGET_LINK_EMUL_SUFFIX "_linux"
diff -Naur gcc-4.2.0.orig/gcc/config/sparc/linux64.h gcc-4.2.0/gcc/config/sparc/linux64.h
--- gcc-4.2.0.orig/gcc/config/sparc/linux64.h	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/sparc/linux64.h	2007-05-18 16:25:23.000000000 -0400
@@ -144,10 +144,10 @@
 
 /* If ELF is the default format, we should not use /lib/elf.  */
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux.so.2"
-#define UCLIBC_DYNAMIC_LINKER32 "/lib/ld-uClibc.so.0"
-#define UCLIBC_DYNAMIC_LINKER64 "/lib/ld64-uClibc.so.0"
+#define GLIBC_DYNAMIC_LINKER32 "/tools/lib32/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER64 "/tools/lib/ld-linux.so.2"
+#define UCLIBC_DYNAMIC_LINKER32 "/tools/lib/ld-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKER64 "/tools/lib/ld64-uClibc.so.0"
 #if UCLIBC_DEFAULT
 #define CHOOSE_DYNAMIC_LINKER(G, U) "%{mglibc:%{muclibc:%e-mglibc and -muclibc used together}" G ";:" U "}"
 #else
@@ -167,7 +167,7 @@
   { "link_arch_default", LINK_ARCH_DEFAULT_SPEC },	  \
   { "link_arch",	 LINK_ARCH_SPEC },
 
-#define LINK_ARCH32_SPEC "-m elf32_sparc -Y P,/usr/lib %{shared:-shared} \
+#define LINK_ARCH32_SPEC "-m elf32_sparc -Y P,/tools/lib32 %{shared:-shared} \
   %{!shared: \
     %{!ibcs: \
       %{!static: \
@@ -176,7 +176,7 @@
         %{static:-static}}} \
 "
 
-#define LINK_ARCH64_SPEC "-m elf64_sparc -Y P,/usr/lib64 %{shared:-shared} \
+#define LINK_ARCH64_SPEC "-m elf64_sparc -Y P,/tools/lib %{shared:-shared} \
   %{!shared: \
     %{!ibcs: \
       %{!static: \
@@ -257,7 +257,7 @@
 #else /* !SPARC_BI_ARCH */
 
 #undef LINK_SPEC
-#define LINK_SPEC "-m elf64_sparc -Y P,/usr/lib64 %{shared:-shared} \
+#define LINK_SPEC "-m elf64_sparc -Y P,/tools/lib %{shared:-shared} \
   %{!shared: \
     %{!ibcs: \
       %{!static: \
diff -Naur gcc-4.2.0.orig/gcc/config/sparc/linux.h gcc-4.2.0/gcc/config/sparc/linux.h
--- gcc-4.2.0.orig/gcc/config/sparc/linux.h	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/sparc/linux.h	2007-05-18 16:24:24.000000000 -0400
@@ -121,8 +121,8 @@
 
 /* If ELF is the default format, we should not use /lib/elf.  */
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
-#define UCLIBC_DYNAMIC_LINKER "/lib/ld-uClibc.so.0"
+#define GLIBC_DYNAMIC_LINKER "/tools/lib/ld-linux.so.2"
+#define UCLIBC_DYNAMIC_LINKER "/tools/lib/ld-uClibc.so.0"
 #if UCLIBC_DEFAULT
 #define CHOOSE_DYNAMIC_LINKER(G, U) "%{mglibc:%{muclibc:%e-mglibc and -muclibc used together}" G ";:" U "}"
 #else
@@ -133,7 +133,7 @@
 
 
 #undef  LINK_SPEC
-#define LINK_SPEC "-m elf32_sparc -Y P,/usr/lib %{shared:-shared} \
+#define LINK_SPEC "-m elf32_sparc -Y P,/tools/lib32 %{shared:-shared} \
   %{!mno-relax:%{!r:-relax}} \
   %{!shared: \
     %{!ibcs: \
diff -Naur gcc-4.2.0.orig/gcc/config/xtensa/linux.h gcc-4.2.0/gcc/config/xtensa/linux.h
--- gcc-4.2.0.orig/gcc/config/xtensa/linux.h	2007-05-16 19:21:19.000000000 -0400
+++ gcc-4.2.0/gcc/config/xtensa/linux.h	2007-05-18 16:23:31.000000000 -0400
@@ -43,7 +43,7 @@
   %{mlongcalls:--longcalls} \
   %{mno-longcalls:--no-longcalls}"
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER "/tools/lib/ld.so.1"
 
 #undef LINK_SPEC
 #define LINK_SPEC \
